pkg/services/ngalert/store/alert_rule.go

```diff
@@ -679,3 +692,39 @@ func (st DBstore) validateAlertRule(alertRule ngmodels.AlertRule) error {
 	}
 	return nil
 }
+
+// deleteOldAlertRuleVersions deletes old alert_rule_version DB records for a particular rule in an org
+func (st *DBstore) deleteOldAlertRuleVersions(ctx context.Context, ruleUID string, orgID, limit int64) (int64, error) {
+	if limit < 1 {
+		return 0, fmt.Errorf("failed to delete old alert rule versions: limit is set to '%d' but needs to be > 0", limit)
+	}
+
+	var affectedRows int64
+	err := st.SQLStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		res, err := sess.Exec(`
+			DELETE FROM
+				alert_rule_version
+			WHERE
+				(rule_org_id = ? AND rule_uid = ?)
+			AND
+				id NOT IN (
```

Comment:
Any reason we're taking this approach rather than the similar one in https://github.com/grafana/grafana/blob/58f6648505d1fbc05d6fc9ee0b37720a9f3c3f5d/pkg/services/ngalert/store/alertmanager.go#L212? Not that I think it's particularly a problem, but if there's a reason we should note it, otherwise I'd favour keeping them in-line.

Reply:
Actually, this was the original approach in the mentioned file as well. I was looking at the older [PR](https://github.com/grafana/grafana/pull/51497/commits/2bfc5310f8c1c3fa9cfbecda62a23c6be4ff3c90#diff-3f90b29bd5d57a76eb6f5c64a90438107da36427d74cafa7a6b5f12f69ea8b15) for reference. Will update this for consistency across the code base.